llvm/lib/Target/AVR/AVRInstrInfo.cpp

```diff
@@ -573,7 +573,7 @@ void AVRInstrInfo::insertIndirectBranch(MachineBasicBlock &MBB,
   if (STI.hasJMPCALL())
     BuildMI(&MBB, DL, get(AVR::JMPk)).addMBB(&NewDestBB);
   else
-    report_fatal_error("cannot create long jump without FeatureJMPCALL");
+    BuildMI(&MBB, DL, get(AVR::RJMPk)).addMBB(&NewDestBB);
```

Comment:
Should we check the BrOffset? Because relative jump can only handle the range between -2k to+2k.

Reply:
No need to do so. The offset is sure to exceed, and incorrect is generated, and the report of "out of range" will be postponed to the assembler or the linker.